ddtrace/contrib/openai/patch.py

```diff
@@ -0,0 +1,540 @@
+            elif prompt:
+                span.set_tag_str("request.prompt", integration.trunc(prompt))
+
+        self._record_request(span, kwargs)
+
+        resp, error = yield
+
+        if resp and not kwargs.get("stream"):
+            if "choices" in resp:
+                choices = resp["choices"]
+                span.set_tag("response.choices.num", len(choices))
+                for choice in choices:
+                    idx = choice["index"]
+                    if "finish_reason" in choice:
+                        span.set_tag_str("response.choices.%d.finish_reason" % idx, str(choice["finish_reason"]))
+                    if "logprobs" in choice:
+                        span.set_tag_str("response.choices.%d.logprobs" % idx, "returned")
+                    if sample_pc_span:
+                        span.set_tag_str("response.choices.%d.text" % idx, integration.trunc(choice.get("text")))
+            span.set_tag("response.object", resp["object"])
+            integration.record_usage(span, resp.get("usage"))
+            if integration.is_pc_sampled_log(span):
+                prompt = kwargs.get("prompt", "")
+                integration.log(
+                    span,
+                    "info" if error is None else "error",
+                    "sampled completion",
+                    attrs={
+                        "prompt": prompt,
+                        "choices": resp["choices"] if resp and "choices" in resp else [],
+                    },
+                )
+
+
+class _ChatCompletionHook(_BaseCompletionHook):
+    _request_tag_attrs = [
+        "temperature",
+        "top_p",
+        "n",
+        "stream",
+        "stop",
+        "max_tokens",
+        "presence_penalty",
+        "frequency_penalty",
+        "logit_bias",
+        "user",
+    ]
+
+    def handle_request(self, integration, span, args, kwargs):
+        sample_pc_span = integration.is_pc_sampled_span(span)
+        messages = kwargs.get("messages")
+        if sample_pc_span and messages:
+
+            def set_message_tag(m):
+                content = integration.trunc(m.get("content", ""))
+                role = integration.trunc(m.get("role", ""))
+                span.set_tag_str("request.messages.%d.content" % idx, content)
+                span.set_tag_str("request.messages.%d.role" % idx, role)
+
+            if isinstance(messages, list):
+                for idx, message in enumerate(messages):
+                    set_message_tag(message)
+            else:
+                set_message_tag(messages)
+
+        self._record_request(span, kwargs)
+
+        resp, error = yield
+
+        completions = ""
+        if resp and not kwargs.get("stream"):
+            if "choices" in resp:
+                choices = resp["choices"]
+                completions = choices
+                span.set_tag("response.choices.num", len(choices))
+                for choice in choices:
```

Comment:
```suggestion
        choices = []
        if resp and not kwargs.get("stream"):
            choices = resp.get("choices", [])
            span.set_tag("response.choices.num", len(choices))
            for choice in choices:
```

Slightly simpler, wdyt?

(would require unindented L459-469, and replacing `completions` below with `choices`)